### PR TITLE
feat(server): clean up interrupted upload files

### PR DESCRIPTION
**Pull request by alextran1502**

### CODE REVIEW

**Review by danieldietzler — APPROVED** [review]
It would be cool if we could write an e2e test for this. Not sure if it's (easily) doable though

File: server/src/middleware/file-upload.interceptor.ts [hunk 1]
```diff
@@ -141,6 +143,14 @@ export class FileUploadInterceptor implements NestInterceptor {
 
   private handleFile(request: AuthRequest, file: Express.Multer.File, callback: Callback<Partial<ImmichFile>>) {
     (file as ImmichMulterFile).uuid = randomUUID();
+
+    request.on('error', (error) => {
+      if (error.message === 'aborted') {
+        this.handleAbortedRequest(request, file);
+        return;
+      }
+    });
```
Comment (jrasm91) [inline server/src/middleware/file-upload.interceptor.ts:152]: This should call `assetService.onUploadError(...)` or similar. Can you move the implementation to that service and add a test for it?

File: server/src/middleware/file-upload.interceptor.ts [hunk 2]
```diff
@@ -141,6 +143,14 @@ export class FileUploadInterceptor implements NestInterceptor {
 
   private handleFile(request: AuthRequest, file: Express.Multer.File, callback: Callback<Partial<ImmichFile>>) {
     (file as ImmichMulterFile).uuid = randomUUID();
+
+    request.on('error', (error) => {
+      if (error.message === 'aborted') {
```
Comment (bo0tzz) [inline server/src/middleware/file-upload.interceptor.ts:148]: We should probably clean up on any kind of error, no?

File: server/src/middleware/file-upload.interceptor.ts [hunk 3]
```diff
@@ -141,13 +124,19 @@ export class FileUploadInterceptor implements NestInterceptor {
...
+      return;
+    });
+
     if (!this.isAssetUploadFile(file)) {
       this.defaultStorage._handleFile(request, file, callback);
       return;
     }
 
     const hash = createHash('sha1');
-    file.stream.on('data', (chunk) => hash.update(chunk));
```
Comment (danieldietzler) [inline server/src/middleware/file-upload.interceptor.ts:150]: Where did that go?

Comment (alextran1502) [reply server/src/middleware/file-upload.interceptor.ts:150]: FAK

**Comment by alextran1502** [comment]
@jrasm91 Done, is this what you are looking for?

**Review by jrasm91 — APPROVED** [review]
Perfect, ty!

**Comment by akostadinov** [comment]
How can one clean the existing failed uploads?

**Comment by bo0tzz** [comment]
Manually

**Comment by akostadinov** [comment]
@bo0tzz , but how exactly to identify what is stale objects from something that is going on right now?

**Pull request merged**